deps/spidershim/src/accessor.h

```diff
@@ -101,6 +107,30 @@ struct CallbackTraits<typename XerTraits<N>::Setter, N> {
   static const unsigned nargs = 0;
 };
 
+struct SignatureChecker {
+  // Performs a signature check if needed.  Returns true and a holder object
+  // unless the signature check fails.
+  static bool CheckSignature(JS::HandleObject accessorData,
+                             v8::Local<Object> _this,
+                             v8::Local<Object>& holder) {
+    Isolate* isolate = Isolate::GetCurrent();
+    JSContext* cx = JSContextFromIsolate(isolate);
+    AutoJSAPI jsAPI(cx, GetObject(_this));
```

Comment:
Are we always guaranteed that `_this` and `accessorData` are same-compartment?  I guess we are because the `accessorData` is in the compartment of the accessor function by construction and `_this` is in that same compartment, ok...